ultra/ultra/baselines/agent_spec.py

```diff
@@ -60,11 +60,16 @@ def __new__(
         task=None,
         max_episode_steps=1200,
         experiment_dir=None,
+        agent_id="",
     ):
         if experiment_dir:
-            print(f"LOADING SPEC from {experiment_dir}/spec.pkl")
-            with open(f"{experiment_dir}/spec.pkl", "rb") as input:
-                spec = dill.load(input)
+            print(
+                f"Loading spec for {agent_id} from {experiment_dir}/agent_metadata.pkl"
+            )
+            with open(f"{experiment_dir}/agent_metadata.pkl", "rb") as metadata_file:
+                agent_metadata = dill.load(metadata_file)
+                spec = agent_metadata["agent_specs"][agent_id]
```

Comment:
I think we discussed this in the ultra meeting, but just affirming that the older experiment files would not be compatible with the new multi-agent structure and how spec is pickled

Reply:
Yes exactly. If one had an experiment that they had run with ULTRA < 0.2 and wanted to perform evaluation on it (python ultra/evaluate.py ...), then it would not work with this new version. This new version's evaluate.py looks for an agent_metadata.pkl file in the experiment directory that contains everything the old spec.pkl did, but also more information (such as the policy class string associated with the agent(s) in the experiment) that was not included, but is helpful.

Reply:
Do we think we should mention this in the documentation or comment it in the evaluate file? I do not know how many people are using ultra at this moment but I dont think it is alot, but nevertheless a warning of some type would be great

Reply:
Oh yah good call. I will look into updating the documentation and then probably add it there.